python/astro_metadata_translator/tests.py

```diff
@@ -149,8 +151,29 @@ def assertObservationInfoFromYaml(self, file, dir=None, check_wcs=True,  # noqa:
             inconsistent.
         """
         header = read_test_file(file, dir=dir)
-        self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
-                                   wcs_params=wcs_params, **kwargs)
+
+        try:
+            self.assertObservationInfo(header, filename=file, check_wcs=check_wcs,
+                                       wcs_params=wcs_params, **kwargs)
+        except AssertionError as error:
+            raise AssertionError("ObservationInfo derived from an dict-like "
+                                 "type is inconsistent.") from error
+
+        # Explicitly verify consistency when instantiating from Header
+        if not isinstance(header, dict):
+            header = header.toDict()
+        # Header can not parse multi-line header keys as lists
+        for key, val in header.items():
+            if isinstance(val, list):
+                header[key] = "".join(val)
```

Comment:
This can't be right. By definition FITS headers can have multiple entries for the same key so this should convert as multiple repeat keys. If you want to be pedantic about it you should handle blank lines and comments/history explicitly with `add_comment` etc methods.

I think it would be more accurate to do something like:

```python
header = Header()
for key, val in header.items():
    if key in ("COMMENT", "HISTORY", ""):
        # comment cards are not relevant for header translation
        continue
    values = val if isinstance(val, list) else [val]
    for v in values:
        header.append((key, val))
```
I haven't tested this but I hope it gives you the idea.

Reply:
Thanks, this was very helpful. What we can't do is
```
Header({"SIMPLE": True, "EXTEND": True, "COMMENT": ["A ", "list"]})
```
so as long as we unravel the list into individual elements Astropy will handle insertions for us (including special treatment for special keywords like COMMENT, HIERARCH, blank etc.) I just got too stuck into typing the incoming object that I didn't simplify.